components/script/dom/element.rs

```diff
@@ -909,16 +909,16 @@ impl Element {
                 continue;
             }
 
-            // TODO: This should skip elements which are not containing blocks of the element we
-            // are scrolling.
-            let scrolling_box = if let Some(document) = node.downcast::<Document>() {
-                ScrollingBox::Viewport(DomRoot::from_ref(document))
-            } else if let Some(element) = node.downcast::<Element>() {
-                ScrollingBox::Element(DomRoot::from_ref(element))
-            } else {
+            let Some(element) = node.downcast::<Element>() else {
                 continue;
             };
 
+            // TODO: This should skip elements which are not containing blocks of the element we
+            // are scrolling.
+            let scrolling_box = match node.downcast::<Document>() {
+                Some(document) => ScrollingBox::Viewport(DomRoot::from_ref(document)),
+                None => ScrollingBox::Element(DomRoot::from_ref(element)),
+            };
```

Comment:
This code no longer makes sense to me. If `node` is a `Document` then it is not an `Element` and therefore we would have taken the `continue;` above. Since the code previously did not panic when calling `determine_scroll_into_view_position` I wonder if it is even possible for `node` to be a `Document` here.